collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java

```diff
@@ -371,24 +379,33 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
                 }
             }
         }
+        if (element.isRepeatable()) {
+            answers.put(PARENT_KEY, parentKey);
+        }
+        if (hasRepeatableGroups(element)) {
+            answers.put(KEY, key);
```

Comment:
The `KEY` is useful for reconstructing order even when there aren't further nested groups. Also, including `KEY` for all repeats matches Briefcase. So this should go in the prior `if`.